core/arch/aarch64/codec.c

```diff
@@ -2341,7 +2363,8 @@ static inline uint
 encode_opnds_b(byte *pc, instr_t *instr, uint enc)
 {
     uint off;
-    if (instr_num_dsts(instr) == 0 && instr_num_srcs(instr) == 1 &&
+    if (((instr_get_opcode(instr) == OP_bl && instr_num_dsts(instr) == 1) ||
+        instr_num_dsts(instr) == 0) && instr_num_srcs(instr) == 1 &&
```

Comment:
nit, the alignment seems wrong, also I would put instr_num_srcs() a separate line for easy read the condition.